Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Memory/SystemAllocator.h>
+
+#include <QString>
+#include <QWidget>
+
+#include "PropertyEditorAPI.h"
+#endif
+
+
+namespace AzQtComponents
+{
+    class BrowseEdit;
+}
+
+namespace AzToolsFramework
+{
+    // A control for representing an AZ::IO::Path field with
+    // a BrowseEdit whose popup button triggers a native
+    // file dialog
+    class PropertyFilePathCtrl
+        : public QWidget
+    {
+        Q_OBJECT
+    public:
+        AZ_CLASS_ALLOCATOR(PropertyFilePathCtrl, AZ::SystemAllocator, 0);
+
+        PropertyFilePathCtrl(QWidget* parent = nullptr);
+
+        AZ::IO::Path GetFilePath() const;
+        void SetFilter(const QString& filter);
+        void SetProductExtension(const AZStd::string& extension);
+        void SetDefaultFileName(const AZStd::string& fileName);
+  
+    Q_SIGNALS:
+        void FilePathChanged();
+
+    public Q_SLOTS:
+        void SetFilePath(AZ::IO::Path filePath);
+
+    private:
+        AzQtComponents::BrowseEdit* m_browseEdit = nullptr;
+        AZ::IO::Path m_currentFilePath;
+        QString m_filter;
+        AZStd::string m_productExtension;
+        AZStd::string m_defaultFileName;
+
+        void OnOpenButtonClicked();
+        void OnClearButtonClicked();
+    };
+
+    class PropertyFilePathHandler
+        : public QObject
+        , public PropertyHandler<AZ::IO::Path, PropertyFilePathCtrl>
+    {
+        Q_OBJECT
+    public:
+        AZ_CLASS_ALLOCATOR(PropertyFilePathHandler, AZ::SystemAllocator, 0);
+
+        AZ::u32 GetHandlerName() const override;
+
+        // Override to be true so that the user to be able to either specify no handler or "Default"
+        // and this handler will be found based on type match (AZ::IO::Path)
+        bool IsDefaultHandler() const override;
+
+        QWidget* CreateGUI(QWidget* parent) override;
+        void ConsumeAttribute(PropertyFilePathCtrl* GUI, AZ::u32 attrib, PropertyAttributeReader* attrValue, const char* debugName) override;
+        void WriteGUIValuesIntoProperty(size_t index, PropertyFilePathCtrl* GUI, property_t& instance, InstanceDataNode* node) override;
+        bool ReadValuesIntoGUI(size_t index, PropertyFilePathCtrl* GUI, const property_t& instance, InstanceDataNode* node)  override;
+    };
+
+    void RegisterFilePathHandler();
+};
```

Comment:
nit: I don't think the `;` is needed here and it should have a closing namespace comment

Reply:
Fixed! The clang-format added the namespace comment, but didn't remove the `;` but I removed it manually